src/VisualStudio/Core/Def/Implementation/LanguageClient/RazorLanguageClient.cs

```diff
@@ -0,0 +1,118 @@
+{
+    /// <summary>
+    /// Defines the LSP server for Razor C#.  This is separate so that we can
+    /// activate this outside of a liveshare session and publish diagnostics
+    /// only for razor cs files.
+    /// TODO - This can be removed once C# is using LSP for diagnostics.
+    /// </summary>
+    [ContentType(ContentTypeNames.CSharpLspContentTypeName)]
+    [ContentType(ContentTypeNames.VBLspContentTypeName)]
+    [ClientName(ClientName)]
+    [Export(typeof(ILanguageClient))]
+    class RazorLanguageClient : ILanguageClient
+    {
+        private readonly IDiagnosticService _diagnosticService;
+        private readonly LanguageServerProtocol _languageServerProtocol;
+        private readonly Workspace _workspace;
+
+        public const string ClientName = "RazorCSharp";
+
+        /// <summary>
+        /// Gets the name of the language client (displayed to the user).
+        /// </summary>
+        public string Name => ServicesVSResources.Razor_CSharp_Language_Server_Client;
+
+        /// <summary>
+        /// Unused, implementing <see cref="ILanguageClient"/>
+        /// No additional settings are provided for this server, so we do not need any configuration section names.
+        /// </summary>
+        public IEnumerable<string>? ConfigurationSections { get; } = null;
+
+        /// <summary>
+        /// Gets the initialization options object the client wants to send when 'initialize' message is sent.
+        /// See https://microsoft.github.io/language-server-protocol/specifications/specification-3-14/#initialize
+        /// We do not provide any additional initialization options.
+        /// </summary>
+        public object? InitializationOptions { get; } = null;
+
+        /// <summary>
+        /// Unused, implementing <see cref="ILanguageClient"/>
+        /// Files that we care about are already provided and watched by the workspace.
+        /// </summary>
+        public IEnumerable<string>? FilesToWatch { get; } = null;
+
+        public event AsyncEventHandler<EventArgs>? StartAsync;
+
+        /// <summary>
+        /// Unused, implementing <see cref="ILanguageClient"/>
+        /// </summary>
+        public event AsyncEventHandler<EventArgs>? StopAsync { add { } remove { } }
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public RazorLanguageClient(LanguageServerProtocol languageServerProtocol, VisualStudioWorkspace workspace, IDiagnosticService diagnosticService)
+        {
+            _languageServerProtocol = languageServerProtocol;
+            _workspace = workspace;
+            _diagnosticService = diagnosticService;
+        }
+
+        public Task<Connection> ActivateAsync(CancellationToken token)
+        {
+            var (clientStream, serverStream) = FullDuplexStream.CreatePair();
+            _ = new RazorLanguageServer(serverStream, serverStream, _languageServerProtocol, _workspace, _diagnosticService);
```

Comment:
1. i hate that we just toss this on the floor.
2. what happens if ActivateAsync is called multiple times?

Reply:
1.  What should be done with it?  the lsp client just needs the stream connection
2.  It won't, guaranteed by the contract with the lsp client

Reply:
1. Store as a field in the class. Tht way we can see it rooted if we get a dump :)
2. ok, can we throw then?

i.e. store this is a field.  if we get called again, throw that this is invalid.  I hate implicit contracts :)